include/logging/log_output_dict.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_LOGGING_LOG_OUTPUT_DICT_H_
+#define ZEPHYR_INCLUDE_LOGGING_LOG_OUTPUT_DICT_H_
+
+#include <logging/log_output.h>
+#include <logging/log_msg2.h>
+#include <stdarg.h>
+#include <toolchain.h>
+#include <sys/util.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Log message type
+ */
+enum log_dict_output_msg_type {
+	MSG_NORMAL = 0,
+	MSG_DROPPED_MSG = 1,
+};
+
+/**
+ * Output header for one dictionary based log message.
+ */
+struct log_dict_output_normal_msg_hdr_t {
+	uint8_t type;
```

Comment:
have you considered reusing log_msg2 directly. There are 2 bits at the start which are used by mpsc_pbuf. They cannot be modified in the provided buffer since they are still in use but if first word (descriptor) would be copied then they could be reused to indicate message type.

This is rather an optimization proposal not necessary for the first iteration.

Reply:
At first I was planning to use `struct log_msg2` directly, but decided not to. This is simply to ease future maintenance. Any changes (even trivial) to `struct log_msg2` would need to be reflected on the Python parser. Also, any alignment and/or padding done to the struct also needs to be done on the Python side which would create a bit more work during arch bring-up. Possible outcome would be that the Python parser is not updated and dictionary-based logging would then be broken on the new arch. (though, it might be just me being paranoid.)